org.freecadweb.FreeCAD.yaml

```diff
@@ -627,10 +644,14 @@ modules:
           config-opts:
             - -DSHIBOKEN_LIBRARY=shiboken2.cpython-39-x86_64-linux-gnu
             - -DPYSIDE_LIBRARY=pyside2.cpython-39-x86_64-linux-gnu
+            - -DCMAKE_C_FLAGS_INIT=-I/usr/include/x86_64-linux-gnu/python3.9 -I/usr/include/python3.9
+            - -DCMAKE_CXX_FLAGS_INIT=-I/usr/include/x86_64-linux-gnu/python3.9 -I/usr/include/python3.9
```

Comment:
ideally there should be a patch for upstream FreeCAD to actually call Python pkg-config and use that.
(I'm saying this for the future, it's good here for now)

Reply:
I was digging in the FreeCAD source and also in this flatpak recipe, and noticed that the `pybind11` dependency has the same purpose than the `boost` dependency (i.e. python bidings for c++).

So I then tried to not include the `pybind11` module in the flatpak recipe, and only specify the `-DBUILD_FLAT_MESH=ON` flag on FreeCAD module build, as FreeCAD has conditionals in its source code to fallback to `boost` when `pybind11` is not used (i.e. when `FREECAD_USE_PYBIN11` option is let to `OFF`), but I've got this compilation error when building FreeCAD : 

```
FAILED: src/App/CMakeFiles/FreeCADApp.dir/DocumentObserverPython.cpp.o 
/usr/bin/ccache /usr/bin/c++ -DDOCDIR=\"/app/share/doc/FreeCAD\" -DFreeCADApp_EXPORTS -DHAVE_CONFIG_H -DQT_CORE_LIB -DQT_NO_DEBUG -DQT_XML_LIB -DRESOURCEDIR=\"/app/share\" -D_OCC64 -I/run/build/FreeCAD/_flatpak_build -I/run/build/FreeCAD/_flatpak_build/src -I/run/build/FreeCAD/src -I/run/build/FreeCAD/_flatpak_build/src/App -I/usr/include/python3.9 -isystem /usr/include/QtCore -isystem /usr/lib/mkspecs/linux-g++ -isystem /usr/include/QtXml -Wall -Wextra -Wno-write-strings -O2 -g -pipe -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fno-omit-frame-pointer -O2 -g -DNDEBUG -fPIC -pthread -fPIC -std=gnu++11 -MD -MT src/App/CMakeFiles/FreeCADApp.dir/DocumentObserverPython.cpp.o -MF src/App/CMakeFiles/FreeCADApp.dir/DocumentObserverPython.cpp.o.d -o src/App/CMakeFiles/FreeCADApp.dir/DocumentObserverPython.cpp.o -c /run/build/FreeCAD/src/App/DocumentObserverPython.cpp
In file included from /usr/include/c++/11.2.0/iterator:65,
                 from /app/include/boost/iterator/iterator_categories.hpp:23,
                 from /app/include/boost/iterator/iterator_adaptor.hpp:12,
                 from /app/include/boost/iterator/reverse_iterator.hpp:10,
                 from /app/include/boost/signals2/detail/auto_buffer.hpp:21,
                 from /app/include/boost/signals2/connection.hpp:21,
                 from /app/include/boost/signals2/signal.hpp:22,
                 from /app/include/boost/signals2.hpp:19,
                 from /run/build/FreeCAD/src/boost_signals2.hpp:6,
                 from /run/build/FreeCAD/src/App/Application.h:27,
                 from /run/build/FreeCAD/src/App/DocumentObserverPython.cpp:30:
/usr/include/c++/11.2.0/bits/stream_iterator.h:68:49: internal compiler error: Segmentation fault
   68 |       : _M_stream(0), _M_value(), _M_ok(false) {}
      |                                                 ^
0x1886451 diagnostic_impl(rich_location*, diagnostic_metadata const*, int, char const*, __va_list_tag (*) [1], diagnostic_t)
	???:0
0x1886826 internal_error(char const*, ...)
	???:0
0xda002e crash_signal(int)
	???:0
0x921adb ggc_set_mark(void const*)
	???:0
0xb419a4 gt_ggc_mx_gimple_df(void*)
	???:0
0xb41bb3 gt_ggc_mx_function(void*)
	???:0
0x882e70 gt_ggc_mx_lang_tree_node(void*)
	???:0
0x881faa gt_ggc_mx_lang_tree_node(void*)
	???:0
0x881faa gt_ggc_mx_lang_tree_node(void*)
	???:0
0x881faa gt_ggc_mx_lang_tree_node(void*)
	???:0
0x881faa gt_ggc_mx_lang_tree_node(void*)
	???:0
0xb3a8e1 gt_ggc_mx_tree_statement_list_node(void*)
	???:0
0x881f45 gt_ggc_mx_lang_tree_node(void*)
	???:0
0x881faa gt_ggc_mx_lang_tree_node(void*)
	???:0
0x882eca gt_ggc_mx_lang_tree_node(void*)
	???:0
0xb39c81 gt_ggc_mx_vec_tree_va_gc_(void*)
	???:0
0x8839b8 gt_ggc_mx_lang_type(void*)
	???:0
0x882bd5 gt_ggc_mx_lang_tree_node(void*)
	???:0
0x8828f5 gt_ggc_mx_lang_tree_node(void*)
	???:0
0x882bc3 gt_ggc_mx_lang_tree_node(void*)
	???:0
Please submit a full bug report,
with preprocessed source if appropriate.
Please include the complete backtrace with any bug report.
```

So I guess the `BUILD_FLAT_MESH` option really requires `pybind11`.


Alternatively, I also tried to only use `pybind11` (i.e. omit `boost` module in the flatpak recipe), but again, there is a build error :

```
========================================================================
Building module coin in ./.flatpak-builder/build/coin-1
========================================================================
-- The C compiler identification is GNU 11.2.0
-- The CXX compiler identification is GNU 11.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at /usr/share/cmake-3.22/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find Boost (missing: Boost_INCLUDE_DIR)
Call Stack (most recent call first):
  /usr/share/cmake-3.22/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/cmake-3.22/Modules/FindBoost.cmake:2360 (find_package_handle_standard_args)
  CMakeLists.txt:160 (find_package)


-- Configuring incomplete, errors occurred!
See also "/run/build/coin/_flatpak_build/CMakeFiles/CMakeOutput.log".
Error: module coin: Child process exited with code 1
```

So it seems that there is a need for both `boost` and `pybind11` libraries.

Reply:
Actually, I figured out that I have a misfunctionning RAM chip that I just added on my computer, hence the segfault :/

I'll try again w/o the RAM chip to see how it goes.